protostar/starknet/cheaters/contracts.py

```diff
@@ -311,3 +308,49 @@ async def invoke(
                 state=state_copy,
                 general_config=StarknetGeneralConfig(),
             )
+
+    async def send_message_to_l2(
+        self,
+        selector: Selector,
+        from_l1_address: Address,
+        to_l2_address: Address,
+        payload: Optional[CairoOrPythonData] = None,
+    ) -> None:
+        cairo_calldata = await self._transform_payload_from_l1_to_cairo_data_by_addr(
+            l1_address=from_l1_address,
+            contract_address=to_l2_address,
+            payload=payload,
+            selector=selector,
+        )
+        entry_point = ExecuteEntryPoint.create_for_testing(
```

Comment:
This will not work if the contract we're calling is `warp`ed or `roll`ed or `mock`ed. We should add a test for that, and use cheatable entrypoint here.

Reply:
Changed to `CheatableExectureEntryPoint`. Added test case.